### PR TITLE
Breaking the legs of a massive broken monster from hell(bull) quickly.

### PR DESCRIPTION
**Pull request by SplinterGP**
## About The Pull Request
nerfs bull heavily due to its current broken state, its neccessary as a bull doenst even use plasma from one border to other when charging on weeds. ignores armor on charge. runs extremely fast on weeeds.

## Why It's Good For The Game
Bull is a extremely broken xeno and needs a nerf as it can kill marines in up to 3 seconds very easily, this should hopefully tone down a bit of the issues such as no plasma use on weeds and etc.

## Changelog
:cl:
balance: bulls use 80% more plasma when charging.
balance: bulls regen plasma a bit slower on advanced ages.
balance: bulls only starts with basic charge, the headbutt is unlocked at mature and gore at ancient.
balance: bull now has -0.7 slowdown on all ages except -0.8 on ancient, before was -0.7 ramping up to -1.0 on ancient
balance: bulls now deals less damage than it used to on charging.
balance: bulls now needs 6 steps to charge instead of 5.
/:cl:

### CODE REVIEW

**Comment by SplinterGP** [comment]
this should hopefully make bull less broken until a proper fix.

**Comment by MiniMeatwad** [comment]
Bull slower than bullet?

**Review by Hughgent — APPROVED** [review]
code is fine

**Comment by SplinterGP** [comment]
> Bull slower than bullet?

no, later :tm:

**Comment by SplinterGP** [comment]
oh fuck the hell did this commit add that sprote

**Comment by SplinterGP** [comment]
okay crisis reverted

**Comment by MiniMeatwad** [comment]
Bull not cause cardiac arrest and cerebral necrosis whenever it bumps into you?

**Comment by MLGTASTICa** [comment]
Gore being unlocked on ancient is a little bit too much ,  as its a crucial part of their playstyle , perhaps make it at elder ? otherwise everything looks fine.

**Review by TiviPlus — APPROVED** [review]

**Comment by SplinterGP** [comment]
gore aint a crucial part of their playstyle, indeed gore is literal just extra damage, gore is just straight extra damage while the other two charges are more interesting, gore will get reworked into something else :tm:

File: code/modules/mob/living/carbon/xenomorph/castes/bull/castedatum_bull.dm [hunk 1]
```diff
@@ -50,6 +50,12 @@
 	upgrade_name = "Young"
 
 	upgrade = XENO_UPGRADE_ZERO
+	actions = list(
+		/datum/action/xeno_action/xeno_resting,
+		/datum/action/xeno_action/regurgitate,
+		/datum/action/xeno_action/ready_charge/bull_charge,
+		/datum/action/xeno_action/activable/bull_charge,
+	)
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/castes/bull/castedatum_bull.dm:58]: change the basetype datum not the young datum

File: code/modules/mob/living/carbon/xenomorph/charge_crush.dm [hunk 2]
```diff
@@ -207,7 +208,7 @@
 						continue
 					charger.visible_message("<span class='danger'>[charger] runs [victim] over!</span>",
 						"<span class='danger'>We run [victim] over!</span>", null, 5)
-					victim.take_overall_damage(CHARGE_SPEED(src) * 10, blocked = victim.run_armor_check("chest", "melee"))
+					victim.take_overall_damage(CHARGE_SPEED(src) * 10, blocked = victim.run_armor_check("chest", "melee")) //This should not use take_overall_damage, this is a direct damage proc.
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/charge_crush.dm:211]: either fix it or remove this comment

File: code/modules/mob/living/carbon/xenomorph/castes/bull/castedatum_bull.dm (hunk 1 above)
Comment (SplinterGP) [reply code/modules/mob/living/carbon/xenomorph/castes/bull/castedatum_bull.dm:58]: o my bad

File: code/modules/mob/living/carbon/xenomorph/charge_crush.dm (hunk 2 above)
Comment (SplinterGP) [reply code/modules/mob/living/carbon/xenomorph/charge_crush.dm:211]: going to fix in another PR, i'll remove it

Comment (Hughgent) [reply code/modules/mob/living/carbon/xenomorph/charge_crush.dm:211]: Still not removed

**Comment by Tennessee116** [comment]
If bull could get a different sprite that was less cartoonish this PR would be the xeno-messiah.

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**